### PR TITLE
Rename MINHTONs PeerInfo and NetworkInfo to Logical and PhysicalNodeInfo

### LINKED ISSUE #3: Rename NetworkInfo to PhysicalNodeInfo

### LINKED ISSUE #4: Rename PeerInfo to LogicalNodeInfo

### PR DESCRIPTION
**Pull request by ltoenning**
Fixes #3
Fixes #4

# Proposed Changes
With this PR the following classes are renamed:
- ``PeerInfo`` :arrow_right: ``LogicalNodeInfo``
- ``NetworkInfo`` :arrow_right: ``PhysicalNodeInfo``

``LogicalNodeInfo`` should contain information which could change during the runtime of a node, e.g., its level and number within the MINHTON tree.

``PhysicalNodeInfo`` should contain information which does not change during the runtime of a node, e.g., its IP address and port number.

The ``LogicalNodeInfo`` and ``PhysicalNodeInfo`` are both part of ``NodeInfo`` .

This renaming should clarify which information is contained in which part and should avoid confusion between the words "Node" and "Peer".

# Definition of Done (optional)
- [x] Feature is implemented
- [x] No known bugs that stops the correct execution
- [x] CI / CD pipeline passed
- [x] The code guidelines were followed
- [x] Unit and / or Integration tests for the new feature
- [x] New feature was added to the documentation

### CODE REVIEW

File: minhton/src/core/routing_information.cpp [hunk 1]
```diff
@@ -208,21 +210,22 @@ void RoutingInformation::setChild(const minhton::NodeInfo &child, uint16_t posit
...
-    throw std::invalid_argument("The Fanout needs to be set properly in the PeerInfo object");
+    throw std::invalid_argument(
+        "The Fanout needs to be set properly in the LogicalNodeInfo object");
   }
 
   if (!child.isInitialized()) {
     throw std::invalid_argument("Child must be initialized / must exist");
   }
 
   // if there is a difference in network info
```
Comment (palask) [inline minhton/src/core/routing_information.cpp:227]: Could argue that this is not entirely accurate anymore, but the comment is kind of unnecessary anyway.

```suggestion

```

**Review by palask — APPROVED** [review]
Just something small that could be changed in one comment.
Aside from that: Can be merged, nothing controversial in this PR since it's just the renaming and a nicely spotted comment mistake in a CMake file ;)

File: minhton/src/core/routing_information.cpp (hunk 1 above)
Comment (ltoenning) [reply minhton/src/core/routing_information.cpp]: Comment removed.

**Review by palask — APPROVED** [review]

File: minhton/src/core/CMakeLists.txt [hunk 2]
```diff
@@ -26,18 +26,18 @@ target_link_libraries(minhton_core_peer_info
     minhton_core_routing_calculations
 )
 
-# Network Info
-add_library(minhton_core_network_info STATIC network_info.cpp)
-target_include_directories(minhton_core_network_info
+# Physical node info
```
Comment (palask) [inline minhton/src/core/CMakeLists.txt:29]: Should be consistent with line 14

```suggestion
# PhysicalNodeInfo
```

**Review by palask — CHANGES_REQUESTED** [review]

**Review by palask — APPROVED** [review]

**Pull request merged**